src/lcals-kokkos/DIFF_PREDICT-Kokkos.cpp

```diff
@@ -0,0 +1,84 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DIFF_PREDICT.hpp"
+
+#if defined(RUN_KOKKOS)
+#include "common/KokkosViewUtils.hpp"
+#include <iostream>
+
+namespace rajaperf {
+namespace lcals {
+
+void DIFF_PREDICT::runKokkosVariant(VariantID vid,
+                                    size_t RAJAPERF_UNUSED_ARG(tune_idx)) {
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  DIFF_PREDICT_DATA_SETUP;
```

Comment:
Would it be possible to make a kokkos version of the DATA_SETUP macros to avoid having to rewrite the loop bodies with kokkos views?

Reply:
@MrBurmark - I sent an email to discuss this topic.  Unfortunately, I think we're constrained here by our respective designs.